src/app/components/RenderMessageContent.tsx

```diff
@@ -67,38 +68,62 @@ export function RenderMessageContent({
       </UrlPreviewHolder>
     );
   };
+  const content: IImageContent = getContent();
+  const renderCaption = content.filename && content.filename !== content.body;
+  let renderedCaption: React.ReactNode = null;
+  if(renderCaption) {
```

Comment:
maybe we can put it into a function so it doesn't not float around and rendered when called explicitly.